### PR TITLE
EZP-29270: Sub-items sorting order not respected by UDW

### PR DESCRIPTION
**Pull request by tischsoic**
https://jira.ez.no/browse/EZP-29270

### CODE REVIEW

**Review by dew326 — CHANGES_REQUESTED** [review]
I don't like the approach. You are creating clauses map based on fetched locations (it's 50 locations but it can be changed by integrator). This could be done in a simpler way. Just create one sortClauses just before `findLocationsByParentLocationId()` you have all information there. It's too many calculations on each load.

File: src/modules/universal-discovery/components/finder/finder.component.js [hunk 1]
```diff
@@ -181,15 +182,16 @@ export default class FinderComponent extends Component {
      * Handles loading more items for a selected parent location
      *
      * @method onLoadMore
-     * @param {Number} parentLocationId
+     * @param {Object} parentLocation
      * @memberof FinderComponent
      */
-    onLoadMore(parentLocationId) {
+    onLoadMore(parentLocation) {
```
Comment (dew326) [inline src/modules/universal-discovery/components/finder/finder.component.js:188]: Potential BC break, you can add new parameter but you cannot change the existing one.

**Review by dew326 — CHANGES_REQUESTED** [review]

File: src/modules/universal-discovery/components/finder/finder.component.js [hunk 2]
```diff
@@ -391,7 +417,9 @@ FinderComponent.propTypes = {
     contentTypesMap: PropTypes.object,
     preselectedLocation: PropTypes.number,
     allowedLocations: PropTypes.array,
-    isVisible: PropTypes.bool
+    isVisible: PropTypes.bool,
+    sortFieldMappings: PropTypes.object,
```
Comment (sunpietro) [inline src/modules/universal-discovery/components/finder/finder.component.js]: Shouldn't it be required?

File: src/modules/universal-discovery/components/finder/finder.component.js (hunk 1 above)
Comment (lserwatka) [reply src/modules/universal-discovery/components/finder/finder.component.js:188]: Is this a public method or API?

**Review by sunpietro — COMMENTED** [review]
It's not public.

File: src/modules/universal-discovery/components/finder/finder.component.js (hunk 1 above)
Comment (sunpietro) [reply src/modules/universal-discovery/components/finder/finder.component.js:188]: It's not public method. The public API is about existing extension points. This method is not one of them

Comment (lserwatka) [reply src/modules/universal-discovery/components/finder/finder.component.js:188]: OK, so we should be fine.

File: src/modules/universal-discovery/components/finder/finder.component.js (hunk 2 above)
Comment (tischsoic) [reply src/modules/universal-discovery/components/finder/finder.component.js]: Good point.

**Review by sunpietro — APPROVED** [review]

**Review by dew326 — APPROVED** [review]

**Review by barbaragr — APPROVED** [review]

**Pull request merged**